src/server/pegasus_server_impl.cpp

```diff
@@ -90,6 +90,28 @@ pegasus_server_impl::pegasus_server_impl(dsn::replication::replica *r)
         1000,
         "multi-get operation iterate count exceed this threshold will be logged, 0 means no check");
 
+    _multi_get_max_iteration_size =
+        dsn_config_get_value_uint64("pegasus.server",
+                                    "rocksdb_multi_get_max_iteration_size",
+                                    100000000,
+                                    "multi-get operation total key-value size exceed "
+                                    "this threshold will stop iterating rocksdb, 0 means no check");
+
+    _rocksdb_max_iteration_count = dsn_config_get_value_uint64(
+        "pegasus.server",
+        "rocksdb_max_iteration_count",
+        1000,
+        "max iteration count for each rocksdb iterator operation, if exceed this threshold,"
+        "iterator will be stopped");
+
+    _rocksdb_iteration_threshold_ns_in_config =
+        dsn_config_get_value_uint64("pegasus.server",
+                                    "rocksdb_iteration_threshold_ns",
+                                    30000000000,
+                                    "max duration for rocksdb iterator operation if exceed "
```

Comment:
> "max duration for rocksdb iterator operation"

What does "iterator operation" mean? As far as I understand, this threshold is for one scan, not for one iterator operation.

```suggestion
                                    "max duration for handling one pegasus scan request (sortkey_count/multiget/scan)"
```